include/range/v3/range_access.hpp

```diff
@@ -247,6 +255,12 @@ namespace ranges
             (
                 pos.next()
             )
+            template<typename Cur>
+            static RANGES_CXX14_CONSTEXPR auto post_increment(Cur & pos)
+            RANGES_DECLTYPE_AUTO_RETURN
+            (
+                pos.post_increment()
+            )
```

Comment:
Keeping in mind that I also hate the name `post_increment`: it is intentional that `next` and `post_increment` are asymmetric, because they are not completely orthogonal. When the cursor does not have `post_increment` and does have `next`, `next` is used to implement both the prefix and postfix increment operators. If the two cursor operations had nicely symmetric names like `next` and `next(int)` I'm concerned that users will get the impression that they must implement both. I *want* `post_increment` to have a long, crappy name to deter people from implementing it unnecessarily.

That said, I'm not convinced that the long, crappy name approach is the best way to address the problem.